### PR TITLE
Fix clipboard promise rejection on web to avoid crash

### PR DESCRIPTION
**Pull request by robertherber**
# Why

When using `getAsyncString` on the web and there is a promise rejection for whatever reason - it currently crashes the app because the Promise isn't properly returned. This fixes this issue.

# How

By returning the promise

# Test Plan

I've tested this in Chrome and Safari.

### CODE REVIEW

**Review by cruzach — APPROVED** [review]
thanks for the fix!

**Comment by cruzach** [comment]
@robertherber could you add an entry to the [changelog](https://github.com/expo/expo/blob/370478fb05c2b0d068fd2ecef6b79d238cf17ac6/packages/expo-clipboard/CHANGELOG.md)?

**Comment by robertherber** [comment]
@cruzach Of course. Added it now!

File: packages/expo-clipboard/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Unpublished
 
+- Fix clipboard getAsyncString crash on exception.
```
Comment (cruzach) [inline packages/expo-clipboard/CHANGELOG.md:5]:
```suggestion
- Fixed `getStringAsync ` causing crashes on Web when an exception is thrown. ([#12494](https://github.com/expo/expo/pull/12494) by [@ robertherber](https://github.com/robertherber))
```

**Review by cruzach — CHANGES_REQUESTED** [review]
woops- could you revert the changes to `yarn.lock`?

**Comment by robertherber** [comment]
@cruzach Oh, that's weird - I just did a yarn install. Anyways - it's reverted now!

**Review by cruzach — APPROVED** [review]

**Pull request merged**